### ISSUE TITLE
GIOU, CIOU, DIOU assign anchors

### ISSUE THREAD

**Issue by fsx950223** [google/automl#270]
Could GIOU, CIOU, DIOU get better performance in anchor assign or use them as losses?

**Comment by lessw2020** [comment]
I'm pretty confident CIoU will get better results.  ( I will test on PyTorch impl of EffDet D0 this week). 

Yolov4 just released with it and they tested all of the above and found CIoU was the best:  
https://arxiv.org/abs/2004.10934?fbclid=IwAR33rU8gGyRkrGs2mQ2h-CBZ7XpPznXD9HJARy_qYlEyTZ3FVuKa-0oLb30

Would be great if CIoU can be tested/integrated here.

**Comment by mingxingtan** [comment]
Yeah, I will try some IOU loss in the coming days.

@fsx950223 , I happened to see you have a PR or different iou losses (https://github.com/tensorflow/addons/pull/914/files). Are they ready to use? If so, I will try to start with that PR.

Also, would be great if you could help add them to this repo. Thanks!

**Comment by fsx950223** [comment]
> Yeah, I will try some IOU loss in the coming days.
> 
> @fsx950223 , I happened to see you have a PR or different iou losses (https://github.com/tensorflow/addons/pull/914/files). Are they ready to use? If so, I will try to start with that PR.
> 
> Also, would be great if you could help add them to this repo. Thanks!

I test them on my project and get better mAP.

**Comment by lessw2020** [comment]
@fsx950223  - that's great to hear re: improved mAP.  
Did you find CIoU was better than DIoU?

**Comment by fsx950223** [comment]
> @fsx950223  - that's great to hear re: improved mAP.  
> Did you find CIoU was better than DIoU?

Yes. Increase about 1 point.

**Comment by lessw2020** [comment]
That's really great news, thanks @fsx950223 for the info!
